### PR TITLE
Show or hide top bar 

### PR DESCRIPTION
**Pull request by infinyte7**
## Purpose / Description
In fullscreen there is no option to hide top bar. So this PR is for that.

## Fixes
Fixes _Link to the issues._

## Approach
Added boolean value in Sharedpreferences with key ```top_bar```
Then accessed that value in AbstractFlashcardViewer, if false then hide else do nothing.

## How Has This Been Tested?
Tested on Android Emulator 
Change setting, check that for showing top bar
```
1. AnkiDroid->Settings->Reviewing->Show Top Bar
2. Now view changes in reviewing
```

## Learning (optional, can help others)
https://developer.android.com/reference/android/transition/Visibility

## Checklist
_Please, go through these checks before submitting the PR._

- [ x ] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [ x ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ x ] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ x ] You have commented your code, particularly in hard-to-understand areas
- [ x ] You have performed a self-review of your own code

### CODE REVIEW

**Comment by infinyte7** [comment]
Add images,
Screenshot_2020-06-11-19-54-03

Screenshot_2020-06-11-19-54-09

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java [hunk 1]
```diff
@@ -220,6 +220,18 @@ private void initSubscreen(String action, PreferenceContext listener) {
                     startActivity(i);
                     return true;
                 });
+                //show / hide top bar
+                CheckBoxPreference showTopBar = (CheckBoxPreference) screen.findPreference("showTopbar");
+                showTopBar.setOnPreferenceClickListener(preference -> {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java:225]: You shouldn't need to do this, it should be handled internally by the preference if you set the `key`

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java [hunk 2]
```diff
@@ -1654,6 +1655,7 @@ protected SharedPreferences restorePreferences() {
         mPrefWaitQuestionSecond = preferences.getInt("timeoutQuestionSeconds", 60);
         mScrollingButtons = preferences.getBoolean("scrolling_buttons", false);
         mDoubleScrolling = preferences.getBoolean("double_scrolling", false);
+        mPrefShowTopbar = preferences.getBoolean("top_bar",false);
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:1658]:
```suggestion
        mPrefShowTopbar = preferences.getBoolean("top_bar", false);
```

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java:225]: oh nice catch!

**Comment by infinyte7** [comment]
I have updated that and also learned other way to implement this.

**Comment by mikehardy** [comment]
@david-allison-1 just as a maintainer note - if I ever mark something needs second review, and you provide it, do feel free to merge and get that merging muscle trained up :-)

@infinyte7 - this'll be in the next alpha!

**Comment by infinyte7** [comment]
It is not working when option is selected to ```Hide system bars```.
```
1. AnkiDroid->Settings->Reviewing->Fullscreen Mode -> Hide system bars
2. AnkiDroid->Settings->Reviewing->Select Show Top bar
```
It should show top bar but not showing.

It is working fine when ```Action Bar``` showing
I tried to solve it but have not come to any solution.

**Comment by david-allison-1** [comment]
> It is not working when option is selected to `Hide system bars`.
> 
> ```
> 1. AnkiDroid->Settings->Reviewing->Fullscreen Mode -> Hide system bars
> 2. AnkiDroid->Settings->Reviewing->Select Show Top bar
> ```
> 
> It should show top bar but not showing.
> 
> It is working fine when `Action Bar` showing
> I tried to solve it but have not come to any solution.

Probably related:

https://github.com/ankidroid/Anki-Android/blob/c26c2abe7251a945a2f7d9a544b0b4c9e3e57916/AnkiDroid/src/main/java/com/ichi2/compat/CompatV19.java#L47-L72

**Comment by infinyte7** [comment]
How can I use this?

**Comment by david-allison-1** [comment]
I presume you'd want to modify the method to check the SharedPreference, and not hide the top bar depending on whether it's set.

**Comment by infinyte7** [comment]
okay, understood, I am implementing that way. Thanks

**Pull request merged**